src/app/icd/ICDNotifier.h

```diff
@@ -0,0 +1,106 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/AppConfig.h>
+#include <cstdint>
+#include <cstring>
+
+#ifndef ICD_MAX_NOTIFICATION_SUBSCRIBERS
+#define ICD_MAX_NOTIFICATION_SUBSCRIBERS 1
+#endif
+
+class ICDNotify;
+
+namespace chip {
+namespace app {
+
+class ICDNotify
+{
+public:
+    enum class KeepActiveFlags : uint8_t
+    {
+        kCommissioningWindowOpen = 0x01,
+        kFailSafeArmed           = 0x02,
+        kExchangeContextOpen     = 0x03,
+    };
+
+    virtual ~ICDNotify() {}
+    virtual void NotifyNetworkActivity()                              = 0;
+    virtual void KeepActiveRequest(KeepActiveFlags request, bool set) = 0;
```

Comment:
Please add some documentation here - what is KeepActiveRequest supposed to do?

I was trying to review the code and understanding it, and without ICD background I could not actually figure it out.